spec.md

```diff
@@ -535,9 +540,55 @@ message PluginCapability {
     Type type = 1;
   }
 
+  message VolumeExpansion {
```

Comment:
If ONLINE and OFFLINE are mutually exclusive (a plugin cannot indicate that it supports both) then we should mention that restriction very clearly using RFC requirement language. For example: "A plugin SHALL NOT indicate that it supports both `ONLINE` and `OFFLINE` volume expansion."

Reply:
Why do you think ONLINE and OFFLINE are mutually exclusive? I would assume a plugin may support both. For example, a AWS volume can be controller expanded both offline and online.

Reply:
@jdef am I correct in assuming that this concern is resolved?

Reply:
Maybe? I guess I'm wondering about cases where the plugin supports both OFFLINE and ONLINE expansion and a user issues a call to the CO: "please expand volume X". The CO knows what the plugin capabilities are (or it will learn them) and ... it *should* know the last call that it issued to the plugin. Meaning that it should be able to reason about the current state of the volume. And because it should know the current state of the volume, it should know which mode(s) of volume expansion are appropriate to initiate.

I can't help but feel like I'm overlooking something here because this proposal is somewhat complex.